internal/controller/nnf_clientmount_controller.go

```diff
@@ -234,7 +234,7 @@ func (r *NnfClientMountReconciler) changeMount(ctx context.Context, clientMount
 			}
 		}
 	} else {
-		unmounted, err := fileSystem.Unmount(ctx, clientMountInfo.MountPath)
+		unmounted, err := fileSystem.Unmount(ctx, clientMountInfo.MountPath, clientMount.Status.Mounts[index].Ready)
```

Comment:
Two questions:
1. Why bother to call this fileSystem.Unmount() if Ready==True?
2. Reconcilers are built around the idea that they "reconcile"--they always verify and correct their Status. If you're going to deviate from that model, then that ought to be described and justified.

Reply:
3. If the status update that was supposed to set Ready=True hits a stale cache error (I forget the error about the cache being stale), then you'll be back here repeating a real unmount anyway. Will that stuff things up again?